src/Decoder.ts

```diff
@@ -59,6 +62,9 @@ export class Decoder {
   headByte = HEAD_BYTE_REQUIRED;
   readonly stack: Array<StackState> = [];
 
+  // TODO: parameterize this property.
+  readonly cachedKeyDecoder = sharedCachedKeyDecoder;
```

Comment:
I'm thinking about the interface to customize a key decoder, so it's not yet public for now.